### PR TITLE
Add `kubernetes.io/arch` label in worker

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:
This PR adds `kubernetes.io/arch` label in worker pool labels. Doing so will add this label in machinedeployment node-template so it can be used to scale the worker pool from zero.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
I am not adding `kubernetes.io/os` label as there will be no benefit for it, cause as of now only supported os is `linux`.

**Release note**:

```other operator
`kubernetes.io/arch` label can now be used for scaling the worker pools from `0` based on CPU architecture.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Can you explain why there are changes to the validation in this PR (the title/description does not talk about it)?

**Comment by acumino** [comment]
> Can you explain why there are changes to the validation in this PR (the title/description does not talk about it)?

It is a minor change and not related to the main motivation of this PR.

File: pkg/operation/botanist/component/extensions/worker/worker.go [hunk 1]
```diff
@@ -173,6 +173,7 @@ func (w *worker) deploy(ctx context.Context, operation string) (extensionsv1alph
 			labels = map[string]string{}
 		}
 		labels["node.kubernetes.io/role"] = "node"
+		labels["kubernetes.io/arch"] = *workerPool.Machine.Architecture
```
Comment (rfranzke) [inline pkg/operation/botanist/component/extensions/worker/worker.go:176]: The kubelet also sets this field. Can there be any issues (different values set by us compared to kubelet)?

**Review by rfranzke — COMMENTED** [review]
/lgtm
Just a minor question

File: pkg/operation/botanist/component/extensions/worker/worker.go (hunk 1 above)
Comment (acumino) [reply pkg/operation/botanist/component/extensions/worker/worker.go:176]: I tested it locally on AWS shoot with a worker pool of both architectures it was working fine. I didn't check the kubelet code that if it overwrites the label or merges it but in any case, there should not be any issue cause the label we are adding is in line with Kubernetes, so even though if it overwrites all the nodes will have `kubernetes.io/arch` label. And if in the future `Kubernetes` changes something we can adapt it on the Gardener side to align with the Kubernetes label.

Comment (rfranzke) [reply pkg/operation/botanist/component/extensions/worker/worker.go:176]: OK, thanks for checking. Then let's continue.

/approve

**Pull request merged**